### ISSUE TITLE
ignore expression evaluation by using "?." in  ILogger Extensions

### ISSUE THREAD

**Issue by waynebaby** [aspnet/Logging#896]
Currently we can use Extension method to use Login

for example we can use

`logger.LogTrace("Hello Error", ex.GetCustomExceptionInfomation());`

to print detailed information in Trace Level.

but most of time we do not have trace level in our configure. in that case the heavy load method "GetCustomExceptionInfomation()" always act as a useless burden.

what I suggest is using "?." suger.  if we can have an extension method to get a level object, if the object is null, the following expression will not be evalulated in runtime. 

for example:

`logger.GetLevel(LogLevel.Trace)?.("Hello Error", ex.GetCustomExceptionInfomation());`

or

`logger.ForTrace()?.("Hello Error", ex.GetCustomExceptionInfomation());`

In this case, if Trace is not configured in appsettings, the evaluation will be skipped.

Is this in a acceptable manner?  I can implement it if you are interested.

**Comment by pakrym** [comment]
I don't think we are willing to add another logging pattern to the abstractions.

Recommended pattern is using `if(logger.IsEnabled(LogLevel.Trace))` before running expensive operations.

**Comment by waynebaby** [comment]
I am totally agree with " will not add another logging pattern  to the abstractions“   and what I suggest is just an extension method, basically an easier version of `if(logger.IsEnabled(LogLevel.Trace))`.


no change to ILogger or Logger at all.
